main/2025/CG-02-25.md

```diff
@@ -23,10 +23,173 @@ No registration is required for VC meetings. The meeting is open to CG members o
+- Colin Murphy
+- Brendan Dahl
+- Deepti Gandluri
+- Luke Wagner
+- Jakob Kummerow
+- Rahul S Chaphalkar
+- Henrik Edstrom
+- Oscar Spencer
+- Zalim Bashorov
+- Manos Koukoutos
+- Shu-yu Guo
+- B N
+- Matthias Liedtke
+- Chris Woods
+- Heejin Ahn
+- Sebastien Doeraene
+- Ryan Hunt
+- Michael Ficarra
+- Ricky Vetter
+- David Degazio
+- Richard Winterton
+- Sam Clegg
+- Emanuel Ziegler
+- Ben Visness
+- Reid rankin
+- Manos Koukoutos
+
 ### Proposals and discussions
 
+### JS interop API for Custom RTTs [#2](https://github.com/WebAssembly/custom-rtts/issues/2)
+
+TL presenting [slides](https://docs.google.com/presentation/d/1Okj-P5hoqUMbAT4BgPXsQYLho05hHNUo1wlX-KybPIQ/edit?usp=sharing)
+
+RH: clarification: Is it only prototypes that we care about? I assume there’s other things we want to configure, e.g. fields reflected as own properties?
+
+TL: Prototypes are the most important. Calling methods is the most important use case. But also because own properties can be emulated with getters/setters on the prototypes. But yes, prototypes are the most important. If we can get that it would be a good start.
+
+RH: Makes sense. The reason I brought it up is that all the constraints you brought up are likely to also apply to these other integrations as well.
+
+SYG: Can you explain how struct.new_desc_with_host solves the immutable shape problem?
+
+RH: new_desc_with_host initializes the field marked as descriptor, and takes an extern ref that engines that can interpret. It could also be an options bag.
+
+SYG: So this is for programmatically making a new shape, not instances with a shape. So with the current alternatives, the shape is declarative, not programmatic like this?
+
+TL: Yeah, we've talked about how we want a declarative solution. I think we do want some kind of imperative way to do this that we can layer some declarative solution on top of, haven't talked about that. If it is just a custom section that can be lowered or polyfilled onto the imperative thing, or some other solutions.
+
+LW: How bad would it be to allow constexprs to call imports?
+
+TL: Would require a more general notion of compile-time imports. We sort of have that for string builtins.
+
+LW: Can't globals currently refer to values known only at instantiation time?
+
+TL: Yes, good point, a previous declared or imported global.
+
+LW: Otherwise, calling an import form the JS API to construct a descriptor from a prototype seems like a good solution
+
+TL: channeling andreas: a function call is not a constextpr
+
+LW: Neither is an import. But you can say for a fixed value of an import, then the result is constant. Same for the result of a function
+
+TL: The benefit of keeping constexprs constant and pure is that it allows lots of implementation strategies. I don’t know how this is done in practice. If everyone is doing it the same way maybe we could just standardize that.
+
+RH: I think you could still do lazy instantiation. My fear about doing global init is that [... ]
```

Comment:
@eqrion and @conrad-watt I think the note taker missed what you were saying here. I guess it's been a while, but if you can recall, you can suggest a change.